src/components/Theme/src/Theme.vue

```diff
@@ -54,35 +44,47 @@ export default {
 		resolveTheme(data, this.parentTheme, this.theme, this.profile);
 		return data;
 	},
-	computed: {
-		styles() {
-			const { colors } = this;
-
-			return {
-				'--neutral-0': colors['neutral-0'],
-				'--neutral-10': colors['neutral-10'],
-				'--neutral-20': colors['neutral-20'],
-				'--neutral-80': colors['neutral-80'],
-				'--neutral-90': colors['neutral-90'],
-				'--neutral-100': colors['neutral-100'],
-				'--color-background': colors.background,
-				'--color-heading': colors.heading,
-				'--color-text': colors.text,
-				'--color-elevation': colors['color-elevation'],
-				'--color-overlay': colors['color-overlay'],
-			};
-		},
-	},
 	beforeUpdate() {
-		// update theme on prop changes
 		resolveTheme(this.$data, this.parentTheme, this.theme, this.profile);
+		this.applyTheme();
+	},
+	beforeMount() {
+		this.applyTheme();
+	},
+	methods: {
+		applyTheme() {
+			const { theme } = createStitches({ theme: this.theme, prefix: 'maker' });
```

Comment:
I'm assuming this takes the JS theme object and creates a bunch of CSS custom properties from it and adding them to the `:root` scope of the page. is that all it's doing or is it also doing anything else? I read [the docs](https://stitches.dev/docs/api#createstitches) for this function but they're not really clear about what it's actually doing.

Reply:
https://stitches.dev/docs/api#theme-1 this provides more details around this function, the only difference is the `core` library does the same thing being agnostic to the framework.

Reply:
that link isn't that helpful, it doesn't say anything about appending styles to the `:root` or why it does that, i wanna understand the rationale behind throwing stuff into `:root`

Reply:
that's the best way to define global css variables: https://developer.mozilla.org/en-US/docs/Web/CSS/:root

Reply:
that link also isn't helpful, it doesn't explain why defining css vars on `:root` is "the best" or better than defining them anywhere else, also as i explained and requested earlier i don't think we should be polluting the global css scope with maker's css vars